http/handler_test.go

```diff
@@ -521,6 +523,113 @@ func TestHandler_error(t *testing.T) {
 	}
 }
 
+func TestHandler_requestAuth(t *testing.T) {
+	core, _, token := vault.TestCoreUnsealed(t)
+
+	rootCtx := namespace.RootContext(nil)
+	te, err := core.LookupToken(rootCtx, token)
+
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	rWithAuthorization, err := http.NewRequest("GET", "v1/test/path", nil)
+	rWithAuthorization.Header.Set("Authorization", "Bearer "+token)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	rWithVault, err := http.NewRequest("GET", "v1/test/path", nil)
+	rWithVault.Header.Set(consts.AuthHeaderName, token)
```

Comment:
Could this be moved down to after the above error is checked? In case `rWithVault` is nil due to an error?

Reply:
Done